SEED-platform/seed
# Start of API v2

#### Any background context you want to provide?
This PR is the start of the version 2.0 API for SEED.

#### What's this PR do?
This PR adds new Django-Rest-Framework API endpoints, documented with Django-Rest-Swagger.  Right now the v1 API is untouched, along with the front-end connections to those v1 API endpoints.  Thus, the front-end user experience, and existing API user experience should be untouched and fully functional.  New class-based function decorators were added to account for the different function arguments for class based methods.

#### How should this be manually tested?
The front-end user-experience should be exactly as it is now.  The existing v1 api endpoints should pass any existing tests.  Swagger page is available at 127.0.0.1:8000/app/api-docs/.  It should be fully functional for the datasets and organizations endpoints that are available on the page.  The main existing issue with the Swagger page is that it doesn't properly render the complex objects we are returning.  There are open, semi-active, issues in django-rest-swagger related to this topic.  In order to operate and test the Swagger page, you must be logged in.  For security sake, exercising the page logged out, logged in with a non-admin user, and logged in with an admin user, should all be performed.

#### What are the relevant tickets?

#### Definition of Done:
